test/SlowTests/Client/UniqueValues.cs

```diff
@@ -276,18 +276,50 @@ public async Task CompareExchangeShouldBeRemovedFromStorageWhenDbGetsDeleted()
                     Name = "💩"
                 };
                 await store.Operations.SendAsync(new PutCompareExchangeValueOperation<User>("emojis/poo", user, 0));
+
                 var dbName = store.Database;
+                var stats = store.Maintenance.ForDatabase(dbName).Send(new GetDetailedStatisticsOperation());
+
+                Assert.Equal(1, stats.CountOfCompareExchange);
+                await store.Maintenance.Server.SendAsync(new DeleteDatabasesOperation(dbName, hardDelete: false));
```

Comment:
I see why it is easier to delete the compare exchange values also on soft delete, but is it correct?

Reply:
The compare exchange values needs to be deleted when the database is deleted (no matter if its hard or soft delete)

Reply:
Yes, we should always delete these. 
see 
![image](https://user-images.githubusercontent.com/116915/53092764-4b4ae000-351e-11e9-8797-26b6cb769d88.png)